spark/src/main/java/org/apache/iceberg/spark/actions/BaseRewriteDataFilesSparkAction.java

```diff
@@ -0,0 +1,443 @@
+    String desc = jobDesc(infoListPair.first(), totalGroups, infoListPair.second().size(),
+        numGroupsPerPartition.get(infoListPair.first().partition), strategy.name());
+
+    Set<DataFile> addedFiles =
+        withJobGroupInfo(newJobGroupInfo(groupID, desc),
+            () -> strategy.rewriteFiles(groupID, infoListPair.second()));
+
+    completedRewrite.offer(groupID);
+    FileGroupRewriteResult fileGroupResult =
+        new FileGroupRewriteResult(addedFiles.size(), infoListPair.second().size());
+
+    results.put(groupID, Pair.of(infoListPair.first(), fileGroupResult));
+  }
+
+  private Result doExecute(Stream<Pair<FileGroupInfo, List<FileScanTask>>> groupStream,
+      RewriteStrategy strategy, int totalGroups, Map<StructLike, Integer> numGroupsPerPartition) {
+
+    ExecutorService rewriteService = Executors.newFixedThreadPool(maxConcurrentFileGroupActions,
+        new ThreadFactoryBuilder().setNameFormat("Rewrite-Service-%d").build());
+
+    ConcurrentLinkedQueue<String> completedRewrite = new ConcurrentLinkedQueue<>();
+    ConcurrentHashMap<String, Pair<FileGroupInfo, FileGroupRewriteResult>> results = new ConcurrentHashMap<>();
+
+    Tasks.Builder<Pair<FileGroupInfo, List<FileScanTask>>> rewriteTaskBuilder = Tasks.foreach(groupStream.iterator())
+        .executeWith(rewriteService)
+        .stopOnFailure()
+        .noRetry()
+        .onFailure((info, exception) -> {
+          LOG.error("Failure during rewrite process for group {}", info.first(), exception);
+        });
+
+    try {
+      rewriteTaskBuilder
+          .run(infoListPair ->
+              rewriteFiles(infoListPair, totalGroups, numGroupsPerPartition, strategy, completedRewrite, results));
+    } catch (Exception e) {
+      // At least one rewrite group failed, clean up all completed rewrites
+      LOG.error("Cannot complete rewrite, partial progress is not enabled and one of the file set groups failed to " +
+          "be rewritten. Cleaning up {} groups which finished being written.", completedRewrite.size(), e);
+      Tasks.foreach(completedRewrite)
+          .suppressFailureWhenFinished()
+          .run(this::abortFileGroup);
+      throw e;
+    }
+
+    commitOrClean(ImmutableSet.copyOf(completedRewrite));
+
+    return new Result(results.values().stream().collect(Collectors.toMap(Pair::first, Pair::second)));
+  }
+
+  private Result doExecutePartialProgress(Stream<Pair<FileGroupInfo, List<FileScanTask>>> groupStream,
+      RewriteStrategy strategy, int totalGroups, Map<StructLike, Integer> numGroupsPerPartition) {
+
+    ExecutorService rewriteService = Executors.newFixedThreadPool(maxConcurrentFileGroupActions,
+        new ThreadFactoryBuilder().setNameFormat("Rewrite-Service-%d").build());
+
+    ExecutorService committerService = Executors.newSingleThreadExecutor(
+        new ThreadFactoryBuilder().setNameFormat("Committer-Service").build());
+
+    int groupsPerCommit = partialProgressEnabled ?
```

Comment:
Isn't `partialProgressEnabled` always true in this case?

Reply:
Yep left over from the unified execute, i'll clean that up